src/helpers/date.js

```diff
@@ -19,3 +30,47 @@ export const formatTime = (time, format = "h:mm a") => {
 export const dateFromObjectId = (id) => {
     return dayjs(new Date(Number.parseInt(id.slice(0, 8), 16) * 1000));
 };
+
+export const now = (date, unit) => {
+    if (!date) {
+        if (unit) {
+            return dayjs.tz().startOf(unit);
+        }
+
+        return dayjs.tz();
+    }
+
+    if (typeof date === "number") {
+        const timestamp = date <= 2_147_483_647 ? date * 1000 : date;
+        return dayjs.tz(timestamp);
+    }
+
+    if (typeof date === "string") {
+        return dayjs.tz(date);
+    }
+
+    if (isDayjs(date)) {
+        // We do this late because under some conditions this is expensive (see: X2-9122)
+        return date;
+    }
+
+    if (date instanceof Date && !Number.isNaN(date.getTime())) {
+        return dayjs.tz(dateToString(date));
+    }
+
+    return dayjs.tz();
+};
+
+const padNumber = (value) => value.toString().padStart(2, "0");
+
+export const dateToString = (date) => {
+    const dateString = `${date.getFullYear()}-${padNumber(date.getMonth() + 1)}-${padNumber(date.getDate())}`;
+    const timeString = `${padNumber(date.getHours())}:${padNumber(date.getMinutes())}:${padNumber(date.getSeconds())}`;
+    return `${dateString} ${timeString}`;
+};
+
+export const getJSDate = (daysDate, isStartDate = true) => {
```

Comment:
Let's call it `toDate` itself because we know dayjs's `.toDate()` gives us a JS date, so these similar names will be very clear.